packages/abi-coder/src/coders/struct.ts

```diff
@@ -53,6 +59,10 @@ export class StructCoder<TCoders extends Record<string, Coder>> extends Coder<
       const fieldCoder = this.coders[fieldName];
       let decoded;
       [decoded, newOffset] = fieldCoder.decode(data, newOffset);
+      if (newOffset % 8 !== 0) {
+        newOffset += 8 - (newOffset % 8);
+      }
+
```

Comment:
@nedsalk Same thing here; keeping precise notes of these inner changes will be very helpful.